src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerActivityExtensions.cs

```diff
@@ -70,13 +68,31 @@ public static JaegerSpan ToJaegerSpan(this Activity activity)
                 ProcessActivityTagRef);
 
             string peerServiceName = null;
-            if ((activity.Kind == ActivityKind.Client || activity.Kind == ActivityKind.Producer) && jaegerTags.PeerService != null)
+            if (activity.Kind == ActivityKind.Client || activity.Kind == ActivityKind.Producer)
             {
-                // Send peer.service for remote calls.
-                peerServiceName = jaegerTags.PeerService;
+                // If priority = 0 that means peer.service may have already been included in tags
+                var addPeerServiceTag = jaegerTags.PeerServicePriority > 0;
 
-                // If priority = 0 that means peer.service was already included in tags.
-                if (jaegerTags.PeerServicePriority > 0)
+                var hostNameOrIpAddress = jaegerTags.HostName ?? jaegerTags.IpAddress;
+
+                // peer.service has not already been included, but net.peer.name/ip and optionally net.peer.port are present
+                if ((jaegerTags.PeerService == null || jaegerTags.PeerServicePriority > 0)
+                    && hostNameOrIpAddress != null)
+                {
+                    peerServiceName = jaegerTags.Port == default
+                        ? hostNameOrIpAddress
+                        : $"{hostNameOrIpAddress}:{jaegerTags.Port}";
```

Comment:
Do we feel strongly that this string allocation should be avoided? The similar code in the Zipkin exporter has a remote endpoint cache since it instantiates a `ZipkinEndpoint` object. Jaeger does not have the notion of an endpoint class.

Reply:
Really our mission is to be as low-impact to the hosting process as possible. Reducing allocations is one part of that. So I personally feel strongly we should do a follow-up to work in a cache. But I'll admit, I kind of geek-out on these optimizations and have a tendency to overdo it 🤣

Reply:
> I kind of geek-out on these optimizations and have a tendency to overdo it 🤣

One of the reason I'm very much enjoying working with you 😄 

> So I personally feel strongly we should do a follow-up to work in a cache.

If you're cool with this as a follow up, I'd be happy to take this on. I've meant to get an issue submitted for the idea of limiting the cache size with an expiration or something. Maybe make sense to take that on at the same time if we think it's a good idea.

Reply:
> If you're cool with this as a follow up, I'd be happy to take this on. I've meant to get an issue submitted for the idea of limiting the cache size with an expiration or something. Maybe make sense to take that on at the same time if we think it's a good idea.

Sounds good 👍